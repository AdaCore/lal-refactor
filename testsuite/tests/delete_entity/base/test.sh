cp ./main.orig ./main.adb
delete_entity -P default.gpr -S main.adb -L 2 -R 14
echo "--  -L 2 -R 14"
cat ./main.adb
