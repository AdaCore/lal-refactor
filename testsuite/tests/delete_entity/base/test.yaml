title: delete_procedure

description: delete_enity tool test

driver: shell_script
