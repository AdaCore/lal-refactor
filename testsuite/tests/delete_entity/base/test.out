--  main:4:46-4:59 Usage of "To_Be_Deleted" that is not safe to delete.
--  (main.orig) -L 2 -R 14
procedure Main is

   Ref : access procedure (Value : Float) := To_Be_Deleted'Access;
   --  A "not-a-call" reference to the procedure --------^


   procedure Another is
   begin
      null;  --  We replace this call with null;
   end Another;
begin

   --  ^ We delete this call


   return;
end Main;
--  (main2.orig) -L 2 -R 14
procedure Main is

begin
   null;
end Main;
--  (main3.orig) -L 9 -R 14
procedure Main is
   procedure Proc (Value : Float) is
   begin
      if Value > 0.0 then
         Proc (Value - 1.0);
      end if;
   end Proc;


begin
   null;
end Main;
--  (main4.orig) -L 2 -R 14
procedure Main is

begin
   null;
end Main;
--  (main5.orig) -L 3 -R 20
procedure Main is
   protected P is

   private
      State : Boolean;
   end P;

   protected body P is

   end P;

begin
   null;
end Main;
--  (main6.orig) -L 2 -R 14
procedure Main is



begin
   null;
end Main;
--  (main7.orig) -L 8 -R 14
procedure Main is

   generic
   procedure Gen;

   procedure Gen is null;


begin
   null;
end Main;
--  (main8.orig) -L 3 -R 14
procedure Main is


   X : Integer := 0;
begin
   if X in 1 | 2 then
      null;
   end if;
end Main;
--  (main9.orig) -L 2 -R 14
procedure Main is


begin
   null;
end Main;
--  (main_10.orig) -L 2 -R 14
procedure Main is


begin
   null;
end Main;
--  (main_11.orig) -L 2 -R 4
procedure Main is

   A, B : constant := 2;

begin
   null;
end Main;
--  (main_11.orig) -L 4 -R 7
procedure Main is
   To_Be_Deleted : constant := 1;

   A : constant := 2;

begin
   null;
end Main;
--  (main_12.orig) -L 4 -R 13
procedure Main is
   package Pkg is
      type Rec
        (A : Integer) is private;
   private
      type Rec
        (A : Integer) is null record;
   end Pkg;
begin
   null;
end Main;
--  (main_13.orig) -L 3 -R 17
procedure Main is
   package Pkg is
      type Rec  is private;
   private
      type Rec  is null record;
   end Pkg;
begin
   null;
end Main;
